File: typescript/base-hmi-types/src/screens/controls/HmiWebControl.ts
import { HmiProperty } from "../base.js";
import { HmiControlWindowBase } from "../base/HmiControlWindowBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiWebControl extends HmiControlWindowBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiWebControl;
  }

  url?: HmiProperty<string>;
  homeUrl?: HmiProperty<string>;
  showAddressBar?: HmiProperty<boolean>;
  useParameterPlaceholders?: HmiProperty<boolean>;
  navigateBack?: HmiProperty<boolean>;
  navigateForward?: HmiProperty<boolean>;
  stop?: HmiProperty<boolean>;
  refresh?: HmiProperty<boolean>;
}

File: src/BaseHmiTypes/Screens/Controls/HmiWebControl.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public class HmiWebControl : HmiControlWindowBase
{
    public HmiWebControl()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiWebControl;
    }

    public HmiProperty<string>? Url { get; set; }

    public HmiProperty<string>? HomeUrl { get; set; }

    public HmiProperty<bool>? ShowAddressBar { get; set; }

    public HmiProperty<bool>? UseParameterPlaceholders { get; set; }

    public HmiProperty<bool>? NavigateBack { get; set; }

    public HmiProperty<bool>? NavigateForward { get; set; }

    public HmiProperty<bool>? Stop { get; set; }

    public HmiProperty<bool>? Refresh { get; set; }
}
